### PR TITLE
INFINITY-3026: Reenable shutdown test

### PR DESCRIPTION
**Pull request by nickbp**
Handle the case where multiple tasks are colocated on the host being shut down. Replace em all.

### CODE REVIEW

**Comment by gabrielhartmann** [comment]
Two failures to provision clusters, re-kicked the runs.

File: docs/pages/tutorials/kafka-tutorial.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ configs:
 ```
 
 9. Save and exit the `svc.yml` file. It should look similar to the following:
+
```
Comment (gabrielhartmann) [inline docs/pages/tutorials/kafka-tutorial.md]: unintentional changes in this PR?

File: frameworks/cassandra/tests/test_zzzrecovery.py [hunk 2]
```diff
@@ -66,9 +66,12 @@ def test_node_replace_replaces_node():
 # @@@@@@@
 @pytest.mark.sanity
 def test_shutdown_host():
-    replace_task = sdk_tasks.get_task_avoiding_scheduler(
+    candidate_tasks = sdk_tasks.get_tasks_avoiding_scheduler(
         config.SERVICE_NAME, re.compile('^node-[0-9]+-server$'))
-    assert replace_task is not None, 'Could not find a node to shut down'
+    assert len(candidate_tasks) != 0, 'Could not find a node to shut down'
```
Comment (gabrielhartmann) [inline frameworks/cassandra/tests/test_zzzrecovery.py]: if we should never have two pods on the same agent, should we:
```
assert len(candidate_tasks) == 1
```
?

Comment (nickbp) [reply frameworks/cassandra/tests/test_zzzrecovery.py]: This is the list of tasks which aren't colocated with the scheduler. Would expect >=2.

File: frameworks/helloworld/tests/test_zzzrecovery.py [hunk 3]
```diff
@@ -288,18 +288,26 @@ def test_pod_replace():
...
+    log.info('Tasks on host {} to be replaced after shutdown: {}'.format(replace_hostname, replace_tasks))
 
     # Instead of partitioning or reconnecting, we shut down the host permanently
-    sdk_cmd.shutdown_agent(replace_task.host)
+    sdk_cmd.shutdown_agent(replace_hostname)
 
-    sdk_cmd.svc_cli(config.PACKAGE_NAME, config.SERVICE_NAME,
-                    'pod replace {}'.format(replace_pod_name))
+    for task in replace_tasks:
+        pod_name = task.name[:-len('-server')]
```
Comment (gabrielhartmann) [inline frameworks/helloworld/tests/test_zzzrecovery.py]: What does this do?  Don't we want pod names, not task names?

File: docs/pages/tutorials/kafka-tutorial.md (hunk 1 above)
Comment (nickbp) [reply docs/pages/tutorials/kafka-tutorial.md]: Not entirely unintentional, just sneaking it in to fix a formatting problem: http://mesosphere.github.io/dcos-commons/tutorials/kafka-tutorial.html

File: frameworks/cassandra/tests/test_zzzrecovery.py (hunk 2 above)
Comment (gabrielhartmann) [reply frameworks/cassandra/tests/test_zzzrecovery.py]: ahhh, my mistake

File: frameworks/helloworld/tests/test_zzzrecovery.py (hunk 3 above)
Comment (nickbp) [reply frameworks/helloworld/tests/test_zzzrecovery.py]: get pod name for `pod replace` operation: `hello-0-server` => `hello-0`

Comment (gabrielhartmann) [reply frameworks/helloworld/tests/test_zzzrecovery.py]: I see.  Now I can read it.  Thanks.

Comment (gabrielhartmann) [reply frameworks/helloworld/tests/test_zzzrecovery.py]: Should we get a `Set` of pod names before issuing replace calls so we don't call replace on the same pod twice?

Comment (nickbp) [reply frameworks/helloworld/tests/test_zzzrecovery.py]: It shouldn't be a problem here, as there's only 1 task/pod. Could add a sanity check to that effect though.

Comment (elezar) [reply frameworks/helloworld/tests/test_zzzrecovery.py]: Could use rstrip("-server") instead to make the intent clearer. Also safer if the string doesn't contain "-server"

Comment (nickbp) [reply frameworks/helloworld/tests/test_zzzrecovery.py]: doesn't rstrip just strip anything that matches the set?

```
>>> 'hello'.rstrip('lo')
'he'
```

**Comment by nickbp** [comment]
Merging as the failed cassandra tests are unrelated to what's being fixed here.

File: frameworks/helloworld/tests/test_zzzrecovery.py (hunk 3 above)
Comment (elezar) [reply frameworks/helloworld/tests/test_zzzrecovery.py]: My bad. I have been using it wrong!

**Pull request merged**